src/beast/beast/http/README.md

```diff
@@ -0,0 +1,401 @@
+using response = message<false, Body, Allocator>;
+```
+
+The template argument `isRequest` is `true` for HTTP requests and `false`
+for HTTP responses, allowing functions to be overloaded or constrained
+based on the type of message they want to be passed.
+
+The `Body` template argument controls the method used to store information
+necessary for receiving or sending the body, as well as providing customizations
+for the actual writing or parsing process. The customizations are used by the
+implementation to perform the `read`, `write`, `async_read`, and `async_write`
+operations on messages.
+
+Beast.HTTP offers `empty_body`, `string_body`, and `streambuf_body` as common
+choices for the `Body` template argument. User-defined objects that meet the `Body`
+requirements may be implemented for custom implementation strategies.
+
+*Note:* Definitions for member functions associated with a `Body` and the types
+it defines should typically be declared inline so they become part of the calling
+code.
+
+## Concepts
+
+### `Body`
+
+Requirements:
+
+`req` denotes any instance of `prepared_request`.<br>
+`resp` denotes any instance of `prepared_response`.<br>
+`preq` denotes any instance of `parsed_request`.<br>
+
+ expression                 | return | type assertion/note/pre/post-condition
+:-------------------------- |:------ |:--------------------------------------
+`Body::value_type`          |        | The type of the `message::body` member.
+`Body::value_type{}`        |        | `DefaultConstructible`
+`Body::reader`              |        | A type meeting the requirements of `Reader`
+`Body::writer`              |        | A type meeting the requirements of `SinglePassWriter` or `MultiPassWriter`
+`Body::prepare(req)`        |        | Prepare `req` for serialization
+`Body::prepare(resp, preq)` |        | Prepare `resp` for serialization
+
+An instance of `message` will inherit the movability and copyability attribuytes
+of the associated `Body::value_type`. That is to say, if the `Body::value_type`
+is movable, then `message` objects using that body will also be movable.
+
+### `Reader`
+
+The implementation for the HTTP parser will construct the body's corresponding
+`reader` object during the parse process. This customization point allows the
+body to determine the strategy for storing incoming body data.
+
+Requirements:
+
+`X`  denotes a type meeting the requirements of `Reader`.<br>
+`a`  denotes a value of type `X`.<br>
+`m`  denotes a value of type `message const&` where
+       `std::is_same<decltype(m.body), Body::value_type>:value == true`.<br>
+`p`  is any pointer.<br>
+`n`  is a value convertible to `std::size_t`.<br>
+`ec` is a value of type `error_code&`.<br>
+
+ expression         | return | type assertion/note/pre/post-condition
+:------------------ |:------ |:--------------------------------------
+`X a(m);`           |        | `a` is no-throw constructible from `m`.
+`a.write(p, n, ec)` |        | No-throw guarantee
+
```

Comment:
To be consistent with these types of tables in the standard:
In the first row you need `X x(m);` where `x` is an identifier (declared above).
`m` is a const lvalue of type `message` where ...
`ec` is a modifiable lvalue of type `error_code`.